thread/common/thrprof.c

```diff
@@ -470,7 +470,11 @@ omrthread_get_handle(omrthread_t thread)
 		In order to work around the compiler error, we have to reach inside
 		the structure do the dirty work. The handle may not even be correct! */
 	uintptr_t *tempHandle;
-	tempHandle = (uintptr_t *)&(thread->handle.__[0]);
+	#if defined(__GNUC__)
+		tempHandle = (uintptr_t *)&(thread->handle.__);
+	#else
+		tempHandle = (uintptr_t *)&(thread->handle.__[0]);
+	#endif
```

Comment:
This should use the same pattern as in https://github.com/eclipse/omr/pull/7323.

Reply:
So I am seeing https://github.com/Deigue/omr/blob/4401fae20ac9db1255fafb68333e322297d43620/include_core/thrtypes.h#L57 , `handle` is of type `OSTHREAD` 

```
OSTHREAD tempHandle = thread->handle;
return  (uint_ptr)*  <cast> tempHandle
```
But what would the cast look like? Where can I find `OSTHREAD` to see what exactly __ is pulling out and how to cast it properly? "handle" also seemed like too broad a search/grep for `/usr/include`

Reply:
Something like this is what I meant:
```c
    OSTHREAD tempHandle = thread->handle;
    return (uintptr_t)*(unsigned long long *)&tempHandle;
```
For z/OS, `OSTHREAD` is defined in `include_core/unix/thrdsup.h`:
```c
typedef pthread_t OSTHREAD;
```

Reply:
Thanks, confirmed it works and everything compiles, and changed accordingly.